src/rgw/rgw_op.h

```diff
@@ -697,6 +749,22 @@ class RGWPutObj : public RGWOp {
   virtual uint32_t op_mask() { return RGW_OP_TYPE_WRITE; }
 };
 
+class RGWPutObj_Filter : public RGWPutObjDataProcessor
+{
+protected:
+  RGWPutObjDataProcessor* next;
+public:
+  RGWPutObj_Filter(RGWPutObjDataProcessor* next) :
+  next(next){}
+  virtual ~RGWPutObj_Filter() {}
+  virtual int handle_data(bufferlist& bl, off_t ofs, void **phandle, rgw_obj *pobj, bool *again) {
+    return next->handle_data(bl, ofs, phandle, pobj, again);
+  }
+  virtual int throttle_data(void *handle, const rgw_obj& obj, bool need_to_wait) {
+    return next->throttle_data(handle, obj, need_to_wait);
+  }
```

Comment:
please use `override` specifier for these. that's enough to document that the function is virtual, so the `virtual` is optional